src/views/view.c

```diff
@@ -1222,6 +1226,85 @@ int dt_view_image_expose(dt_view_image_over_t *image_over, uint32_t imgid, cairo
   return missing;
 }
 
+void
```

Comment:
I did that at first and it makes the dt_view_image_expose() even more unreadable. So to me it's a compromise between readability and code duplication. Would be good to have other point of view as I've no strong opinion and both solutions have pro and cons.